pkg/telemetryquerylanguage/tql/parser.go

```diff
@@ -149,7 +149,23 @@ func (n *IsNil) Capture(_ []string) error {
 
 type EnumSymbol string
 
-func ParseQueries(statements []string, functions map[string]interface{}, pathParser PathExpressionParser, enumParser EnumParser) ([]Query, error) {
+type Parser struct {
+	functions  map[string]interface{}
+	pathParser PathExpressionParser
+	enumParser EnumParser
+	logger     Logger
+}
+
+func NewParser(functions map[string]interface{}, pathParser PathExpressionParser, enumParser EnumParser, logger Logger) Parser {
```

Comment:
I like this idea.  It would allow the transform processor to do something like `WithLogger` and hide the `common.NewTQLLogger` logic within the Options function implementation.

I think we'd want to keep `functions` and `pathParser` required.  I could go either way with `enumParser`; some contexts may not have enums and wouldn't need to pass in that function.

Switching to Options may be out of scope for this PR tho, which is already quite large.  I am ok allowing this through as is with the expressed intent to switch to Options soon in the future. Since we are still alpha, we can get away with a lot of breaking changes between versions. @kentquirk what do you think?

Reply:
I'd be happy with that. I'm even willing to write that code, and I've just put up #13759 to track it.

Reply:
Sweet thanks